src/lib/dataEntities/transactionRequest.ts

```diff
@@ -0,0 +1,51 @@
+import { TransactionRequest } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+
+/**
+ * Make some properties required
+ */
+type NonOptional<TType, TProps extends keyof TType> =
+  // make the picked props required
+  Required<Pick<TType, TProps>> &
+    // and include all existing properies
+    TType
+
+/**
+ * A transaction request for a transaction that will trigger some sort of
+ * execution on the L2
+ */
+export interface L1ToL2TransactionRequest {
+  txRequest: NonOptional<TransactionRequest, 'to' | 'data' | 'value'>
```

Comment:
They define the minimum for a user interaction - from these fields the others can be derived in different ways> What I wanted to do was have only those fields, but make sure they overlap with a transaction request. I think a better way and clearer to do this would be use `Pick` though - I'll update.